include/librealsense2/hpp/rs_types.hpp

```diff
@@ -96,6 +96,14 @@ struct rs2_update_progress_callback
 };
 typedef std::shared_ptr<rs2_update_progress_callback> rs2_update_progress_callback_sptr;
 
+struct rs2_option_value_update_callback
+{
+    virtual void on_value_changed( const std::vector< rs2_option > & list ) = 0;
```

Comment:
We cannot use vector across the hourglass top/bottom

Reply:
Removed use of vector. Implemented `options_list`.